lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -116,6 +117,51 @@ sub list {
     my ($self) = @_;
 }
 
+sub _load_test_preset ($self, $preset_key) {
+    return undef unless defined $preset_key;
+    # avoid reading INI file again on subsequent calls
+    state %presets;
+    return $presets{$preset_key} if exists $presets{$preset_key};
+    $presets{$preset_key} = undef;
+    # read preset from an INI section [test_presets/…] or fallback to defaults assigned on setup
+    my $config = $self->app->config;
+    return undef unless my $ini_config = $config->{ini_config};
+    my $ini_key = "test_preset $preset_key";
+    return $presets{$preset_key}
+      = $ini_config->SectionExists($ini_key)
```

Comment:
I still don't like that we are creating some extra code for this specific feature to handle nested data, while we could do this in a more generic way in Setup.pm with the group feature.
Also it moves any validation we might want to do to runtime instead of server startup.
Since it's something internal we can probably change the implementation later, but just saying.
Not sure what others think.